src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -28,23 +139,13 @@ export const StudyProposal: React.FC<{ memberId: string }> = ({ memberId }) => {
             })
             router.push(`/researcher/dashboard`)
         },
-        onError(error) {
-            // TODO server action max filesize error doesn't propagate through here...
+        onError: async (error, _, context: { studyId: string; studyJobId: string } | undefined) => {
             console.error(error)
-            notifications.show({ message: String(error), color: 'red' })
-        },
-    })
-
-    const studyProposalForm = useForm<StudyProposalFormValues>({
-        validate: zodResolver(studyProposalSchema),
-        validateInputOnBlur: true,
-        initialValues: {
-            title: '',
-            piName: '',
-            irbDocument: null,
-            descriptionDocument: null,
-            agreementDocument: null,
-            codeFiles: [],
+            if (!context) return
+            const deletePath = pathForStudy({ memberSlug: memberSlug, studyId: context.studyId })
+            const deleteStudyFilesURL = await signedUrlForDeletingStudyFilesAction(deletePath)
+            await fetch(deleteStudyFilesURL)
+            await onDeleteStudyAction({ studyId: context.studyId, studyJobId: context.studyJobId })
```

Comment:
same as above, we ~can't~ shouldn't call actions back-to-back.    Is the delete path different from the upload path though? Seems like it should not be?  If it does need to be, then have the study save action return that as well.

Reply:
yeah disregarding per your later comment, but the fetch needs to happen on the client side so we would need both calls here (just confirming if that sounds right)